packages/service-worker/src/module.ts

```
@@ -115,10 +115,10 @@ export function ngswAppInitializer(
       }
     }
 
-    // Don't return anything to avoid blocking the application until the SW is registered or
-    // causing a crash if the SW registration fails.
+    // Don't return anything to avoid blocking the application until the SW is registered.
+    // Catch and ignore the error if SW registration fails.
     readyToRegister$.pipe(take(1)).subscribe(
-        () => navigator.serviceWorker.register(script, {scope: options.scope}));
+        () => navigator.serviceWorker.register(script, {scope: options.scope}).catch(() => {}));
```

I don't think it is a good idea to swallow that error. If the SW fails to register (for whatever reason), the developer should know.
If you want to avoid that the uncaught rejection error, maybe we could explicitly pass it to `console.error()` inside the `catch()`? WDYT?

---

Hey @gkalpak, thanks for the feedback.
It's a good idea, here is the diff I just push in my commit:
```patch
--- a/packages/service-worker/src/module.ts
+++ b/packages/service-worker/src/module.ts
@@ -118,7 +118,9 @@ export function ngswAppInitializer(
     // Don't return anything to avoid blocking the application until the SW is registered.
     // Catch and ignore the error if SW registration fails.
     readyToRegister$.pipe(take(1)).subscribe(
-        () => navigator.serviceWorker.register(script, {scope: options.scope}).catch(() => {}));
+        () =>
+            navigator.serviceWorker.register(script, {scope: options.scope})
+                .catch(err => console.error('NGSW - service worker register failed with: ', err)));
```
PS: Would you be able to help me fixing the CI tests ?
If you think this fix is a good idea of course.

---

The failing tests on CI are due to the fact that we are mocking `serviceWorker.register()` [here][1], but not returning a promise (which was not necessary before). You need to change the above line to something like:

```diff
-beforeEach(() => swRegisterSpy = spyOn(navigator.serviceWorker, 'register'));
+beforeEach(() => swRegisterSpy = spyOn(navigator.serviceWorker, 'register').and.returnValue(Promise.resolve()));
```

[1]: https://github.com/angular/angular/blob/957f594d7cb596e495d166db3d3ecb5103abd7ea/packages/service-worker/test/module_spec.ts#L31